modules/services/trust-relationship/main.tf

```diff
@@ -39,18 +39,14 @@ data "aws_iam_policy_document" "trust_relationship" {
 }
 
 resource "aws_iam_role" "cspm_role" {
-  count = var.is_organizational && !var.provision_caller_account ? 0 : 1
-
   name               = var.role_name
   assume_role_policy = data.aws_iam_policy_document.trust_relationship.json
   tags               = var.tags
 }
 
 
 resource "aws_iam_role_policy_attachment" "cspm_security_audit" {
-  count = var.is_organizational && !var.provision_caller_account ? 0 : 1
-
-  role       = aws_iam_role.cspm_role[0].id
+  role       = aws_iam_role.cspm_role.id
```

Comment:
To be safe / consistent, let's make all role creation to be inline and have no separate policy attachment resource.
What do you think ?

Reply:
I don't think its an issue for single install.